### PR TITLE
2472 utilities added functions

### PR DESCRIPTION
**Pull request by stevenburnell-moj**
Refactor existing Utilities tests to make them easier to read.
Add missing tests for newer functions that were not already there.

### CODE REVIEW

File: test/utilities/max_height_test.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+
+describe('utilities.maxHeight', function() {
+  it('should return the maxHeight found in a collection', function() {
+    var $item1 = $("<p id=\"test-maxheight-item1\">width item 1");
+    var $item2 = $("<p id=\"test-maxheight-item2\">width item 2");
+    var $item3 = $("<p id=\"test-maxheight-item3\">width item 3");
+    var $collection = $(); // empty jQuery collection.
+
+    // Add some widths...
```
Comment (chrispymm) [inline test/utilities/max_height_test.js:12]: * heights

File: test/utilities/unique_string_test.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+describe('utilities.uniqueString', function() {
+  it('should return a string without passed arguments', function() {
+    expect(utilities.uniqueString().constructor).to.equal(String().constructor);
+    expect(typeof utilities.uniqueString()).to.equal("string");
+  });
+
+  it('should return a string that include the passed string value', function() {
+    var str = 'mystringvalue';
+    expect(utilities.uniqueString(str).constructor).to.equal(String().constructor);
+    expect(typeof utilities.uniqueString(str)).to.equal("string");
```
Comment (chrispymm) [inline test/utilities/unique_string_test.js:14]: The test is supposed to be testing `it('should return a string that include the passed string value')` but there doesn't appear to be an expectation that the returned string contains the required string.

File: test/utilities/update_dom_by_api_request_test.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        expect($targetNode).to.exist;
+        expect($targetNode.length).to.equal(1);
+        expect($insertedNode).to.exist;
+        expect($insertedNode.length).to.equal(1);
+        expect($targetNode.find($insertedNode).length > 0).to.be.true;
+      }
+    });
+  });
+
+  it('should place new element before target node', function() {
```
Comment (chrispymm) [inline test/utilities/update_dom_by_api_request_test.js:46]: Think this should be `it('should place new element after target node)` as you have type `after` in the `updateDomByApiRequest` call, and you're expectation is `($targettnode).next()`

File: test/utilities/update_dom_by_api_request_test.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  afterEach(function() {
+    $(document.body).find("#" + insertId).remove();
+  });
+
+  after(function() {
+    $(document.body).find("#" + targetId).remove();
+    $.get = get;
+  });
+
+  it('should place new element after target node', function() {
```
Comment (chrispymm) [inline test/utilities/update_dom_by_api_request_test.js:27]: Should this one be `it('should place new element inside target node')` or maybe `should append new element within target node`?  As the expectation is `expect($targetNode).find($insertedNode)` so we're looking for elements within `$targetNode`

**Review by chrispymm — CHANGES_REQUESTED** [review]

File: test/utilities/max_height_test.js (hunk 1 above)
Comment (stevenburnell-moj) [reply test/utilities/max_height_test.js]: cut-and-paste error now corrected

File: test/utilities/unique_string_test.js (hunk 2 above)
Comment (stevenburnell-moj) [reply test/utilities/unique_string_test.js]: Corrected test to check for included value.

File: test/utilities/update_dom_by_api_request_test.js (hunk 3 above)
Comment (stevenburnell-moj) [reply test/utilities/update_dom_by_api_request_test.js]: Correct. Have fixed typo.

File: test/utilities/update_dom_by_api_request_test.js (hunk 4 above)
Comment (stevenburnell-moj) [reply test/utilities/update_dom_by_api_request_test.js]: Yes, it should be inside/'append to' the target node as it is testing the default, single param, action. Have now corrected.

**Review by chrispymm — APPROVED** [review]

**Pull request merged**